projects/rx-form-mapper/src/lib/services/validator-resolver.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable, InjectFlags, Injector } from '@angular/core';
+import { AbstractControl, AsyncValidatorFn, ValidatorFn } from '@angular/forms';
+import { RxAsyncValidator, RxValidator } from '../types';
+
+@Injectable()
+export class ValidatorResolver {
+
+	public constructor(private readonly injector: Injector) {}
+
+	public resolve(validator: RxAsyncValidator): AsyncValidatorFn;
+	public resolve(validator: RxValidator): ValidatorFn;
+	public resolve(validator: any): any {
+
+		if (this.isValidatorFn(validator)) {
+			return validator;
+		}
+
+		let validatorInstance = null;
+
+		if (this.isValidatorInstance(validator)) {
+			validatorInstance = validator;
+		} else {
+			validatorInstance = this.injector.get(validator as any, null, InjectFlags.Optional);
+			validatorInstance ??= new validator();
+		}
+
+		return (c: AbstractControl) => validatorInstance.validate(c);
+
+	}
+
+	private isValidatorFn(value: any): boolean {
+		return typeof(value) === 'function' && !value.prototype.validate;
```

Comment:
Non potresti usare qualcosa di già definito? tipo da loadash?

Reply:
Non è una cattiva idea, inizialmente volevo evitare di includere dipendenze extra alla libreria ma è abbastanza evidente che la cosa mi sia sfuggita di mano... Provvedo!

Reply:
Si, sai che io sono a favore del non utilizzare le dipendenze esterne.
So però che ti può davvero far utile in questo tipo di progetto.

Includi solo ciò che ti serve da loadash :=)